server/server.js

```diff
@@ -73,80 +73,48 @@ Server.prototype.terminate = function (done) {
 var VERSION_CLIENT_STOREDATA = '0.13.1'
 
 Server.prototype._setup = function (httpServer, configuration) {
-  var engine = DefaultEngineIO
-  var engineConf
-
-  this.subscriber = Core.Persistence.pubsub()
-
-  this.subscriber.on('message', this._handlePubSubMessage.bind(this))
-
-  var oldPublish = Core.Persistence.publish
-
-  // Log all outgoing to redis server
-  Core.Persistence.publish = function (channel, data, callback) {
-    logging.info('#redis.message.outgoing', channel, data)
-    oldPublish(channel, data, callback)
-  }
-
   configuration = configuration || {}
 
-  if (configuration.engineio) {
-    engine = configuration.engineio.module
-    engineConf = configuration.engineio.conf
-
-    this.engineioPath = configuration.engineio.conf ? configuration.engineio.conf.path : 'default'
-  }
-
   this._setupSentry(configuration)
-
-  this.socketServer = engine.attach(httpServer, engineConf)
-  this.socketServer.on('connection', this._onSocketConnection.bind(this))
+  this._setupEngineio(httpServer, configuration.engineio)
+  this._setupDistributor()
```

Comment:
some of these rewrites are not very related to clientSession, although they are good rewrites. I suggest we split the PRs up (just do a  separate PR into this branch, or master and then rebase) so that reviewing is a bit easier.